libr/io/map.c

```diff
@@ -139,8 +140,41 @@ void io_map_calculate_skyline(RIO *io) {
 			last_map = map;
 		}
 	}
-
 	r_binheap_clear (&heap);
+
+	const RPVector *skyline = &io->map_skyline;
+	RPVector *shadow = &io->map_skyline_shadow;
+	RInterval *cur_itv = NULL;
+
+	for (i = 0; i < r_pvector_len (skyline); i++) {
+		const RIOMapSkyline *part = r_pvector_at (skyline, i);
+		ut64 part_from = part->itv.addr;
+		ut64 part_size = part->itv.size;
+
+		if (!cur_itv) {
+			cur_itv = R_NEW (RInterval);
+			cur_itv->addr = part_from;
+			cur_itv->size = part_size;
+			continue;
+		}
+
+		if (part_from == r_itv_end (*cur_itv)) {
+			cur_itv->size += part_size;
+		} else {
+			if (!r_pvector_push (shadow, cur_itv)) {
+				R_FREE (cur_itv);
+				break;
+			}
+			cur_itv = R_NEW (RInterval);
+			cur_itv->addr = part_from;
+			cur_itv->size = part_size;
+		}
+	}
+	if (cur_itv) {
+		if (!r_pvector_push (shadow, cur_itv)) {
+			R_FREE (cur_itv);
+		}
+	}
 out:
 	r_pvector_clear (&events);
```

Comment:
leaking cur_itv?

Reply:
why leaking? it's pushed in the vector and it'll be freed when clearing the vector, am i assuming that right?